rotkehlchen/rotkehlchen.py

```diff
@@ -944,6 +991,20 @@ def set_settings(self, settings: ModifiableDBSettings) -> Tuple[bool, str]:
             if settings.btc_derivation_gap_limit is not None:
                 self.chain_manager.btc_derivation_gap_limit = settings.btc_derivation_gap_limit
 
+            if settings.current_price_oracles is not None:
```

Comment:
if you do what I told you in encoding an enum value would come here and not a string. And this should make things easier in all these functions.